tests/e2e_tests/conftest.py

```diff
@@ -67,10 +68,11 @@ def authenticator_fund_config(request: FixtureRequest) -> TestFundConfig:
 
 
 @pytest.fixture
-def context(request: FixtureRequest, browser: Browser, e2e_test_secrets: EndToEndTestSecrets):
+def context(new_context: CreateContextCallback, request: FixtureRequest, e2e_test_secrets: EndToEndTestSecrets):
     e2e_env = request.config.getoption("e2e_env")
     http_credentials = e2e_test_secrets.HTTP_BASIC_AUTH if e2e_env in {"dev", "test"} else None
-    return browser.new_context(http_credentials=http_credentials)
+    context = new_context(http_credentials=http_credentials)
+    yield context
```

Comment:
question: any difference between `yield` and `return` is this context? or was `return` not closing the browser context correctly?

Reply:
No different in this context. `yield` is useful in fixtures if you want to do things during test teardown (ie after returning the fixture).

This could've stayed as a `return`. I made it a `yield` because at some point while making this I added some cleanup to the fixture after the `yield`.

When I removed that cleanup I forgot to turn this back into a `return` (and also it doesn't matter)

Reply:
(ive made it a return again)